<html>

<head>
<meta http-equiv=Content-Type content="text/html; charset=windows-1252">
<meta name=Generator content="Microsoft Word 15 (filtered)">
<style>
<!--
 /* Font Definitions */
 @font-face
	{font-family:Wingdings;
	panose-1:5 0 0 0 0 0 0 0 0 0;}
@font-face
	{font-family:"Cambria Math";
	panose-1:2 4 5 3 5 4 6 3 2 4;}
@font-face
	{font-family:Calibri;
	panose-1:2 15 5 2 2 2 4 3 2 4;}
@font-face
	{font-family:Cambria;
	panose-1:2 4 5 3 5 4 6 3 2 4;}
@font-face
	{font-family:Tahoma;
	panose-1:2 11 6 4 3 5 4 4 2 4;}
 /* Style Definitions */
 p.MsoNormal, li.MsoNormal, div.MsoNormal
	{margin-top:0cm;
	margin-right:0cm;
	margin-bottom:10.0pt;
	margin-left:0cm;
	line-height:115%;
	font-size:11.0pt;
	font-family:"Calibri",sans-serif;}
h1
	{mso-style-link:"Heading 1 Char";
	margin-top:12.0pt;
	margin-right:0cm;
	margin-bottom:3.0pt;
	margin-left:0cm;
	line-height:115%;
	page-break-after:avoid;
	font-size:16.0pt;
	font-family:"Cambria",serif;}
h2
	{mso-style-link:"Heading 2 Char";
	margin-top:12.0pt;
	margin-right:0cm;
	margin-bottom:3.0pt;
	margin-left:0cm;
	line-height:115%;
	page-break-after:avoid;
	font-size:14.0pt;
	font-family:"Cambria",serif;
	font-style:italic;}
a:link, span.MsoHyperlink
	{color:blue;
	text-decoration:underline;}
p.MsoNoSpacing, li.MsoNoSpacing, div.MsoNoSpacing
	{margin:0cm;
	font-size:11.0pt;
	font-family:"Calibri",sans-serif;}
span.Heading1Char
	{mso-style-name:"Heading 1 Char";
	mso-style-link:"Heading 1";
	font-family:"Cambria",serif;
	font-weight:bold;}
span.Heading2Char
	{mso-style-name:"Heading 2 Char";
	mso-style-link:"Heading 2";
	font-family:"Cambria",serif;
	font-weight:bold;
	font-style:italic;}
.MsoChpDefault
	{font-family:"Calibri",sans-serif;}
 /* Page Definitions */
 @page WordSection1
	{size:595.3pt 841.9pt;
	margin:70.85pt 70.85pt 2.0cm 70.85pt;}
div.WordSection1
	{page:WordSection1;}
 /* List Definitions */
 ol
	{margin-bottom:0cm;}
ul
	{margin-bottom:0cm;}
-->
</style>

</head>

<body lang=EN-GB link=blue vlink="#954F72" style='word-wrap:break-word'>

<div class=WordSection1>

<h1 style='margin-top:0cm;margin-right:0cm;margin-bottom:6.0pt;margin-left:
0cm;text-align:justify'><span lang=EN-US style='line-height:115%;font-family:
"Calibri",sans-serif'>Manual: Hapl-o-Mat via Hapl-o-MatGUI</span></h1>

<p class=MsoNormal style='margin-bottom:12.0pt'><span lang=EN-US>Please also
see the README.</span></p>

<p class=MsoNormal style='text-align:justify'><span lang=EN-US>Hapl-o-Mat is a software
for HLA haplotype inference coded in C++. Besides estimating haplotype
frequencies via an expectation-maximization algorithm, it is capable of
processing HLA genotype population data. This includes translation of alleles
between various typing resolutions and resolving allelic and genotypic
ambiguities. </span></p>

<p class=MsoNormal style='text-align:justify'><span lang=EN-US>For more
information refer to our publication on Hapl-o-Mat:</span></p>

<p class=MsoNormal style='text-align:justify'><span lang=EN-US>Schaefer C,
Schmidt AH, Sauter J. Hapl-o-Mat: open-source software for HLA haplotype
frequency estimation from ambiguous and heterogeneous data. BMC Bioinformatics.
2017;18(1):284. Published 2017 May 30. doi:10.1186/s12859-017-1692-y</span></p>

<p class=MsoNormal style='text-align:justify'><span lang=EN-US>If you use
Hapl-o-Mat for your research, please cite preferably the journal article.</span></p>

<p class=MsoNormal style='text-align:justify'><span lang=EN-US>&nbsp;</span></p>

<h2 style='margin-top:6.0pt;margin-right:0cm;margin-bottom:12.0pt;margin-left:
0cm'><span lang=EN-US style='font-family:"Calibri",sans-serif;font-style:normal'>System
requirements and installation </span></h2>

<p class=MsoNormal style='text-align:justify'><span lang=EN-US>After successful
installation, Hapl-o-Mat can be executed both from command line or in the
Hapl-o-MatGUI. </span></p>

<p class=MsoNormal style='margin-bottom:6.0pt'><b><span lang=EN-US>Installation
on LinuxOS/macOS: </span></b></p>

<p class=MsoNormal style='margin-bottom:6.0pt'><span lang=EN-US>In order to use
Hapl-o-Mat with the GUI on Linux or macOS, please install both, Hapl-o-Mat C++
source code (1) and Hapl-o-MatGUI python3 source code (2).</span></p>

<p class=MsoNormal style='margin-bottom:6.0pt'><b><span lang=EN-US>Installation
on Windows OS:</span></b></p>

<p class=MsoNormal style='margin-bottom:6.0pt'><span lang=EN-US>For the use of
Hapl-o-Mat with the GUI on Windows please install the Hapl-o-Mat Windows binary
version (3). This version contains both, Hapl-o-Mat and the GUI and has no
further system requirements.</span></p>

<p class=MsoNormal style='margin-bottom:12.0pt'><span lang=EN-US>Alternatively,
you can install and use Hapl-o-Mat C++ source code (1) and Hapl-o-MatGUI
python3 source code (2) on your Windows device.</span></p>

<p class=MsoNormal style='margin-left:21.3pt;text-indent:-21.3pt'><b><span
lang=EN-US>(1)<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></b><b><span lang=EN-US>Hapl-o-Mat C++ source code:</span></b><span
lang=EN-US> For system requirements and installation of Hapl-o-Mat see the
respective tutorials on GitHub (<a href="https://github.com/DKMS/Hapl-o-Mat">https://github.com/DKMS/Hapl-o-Mat</a>),
e.g. &quot;detailedGettingStartedWindows.pdf&quot; and &quot;detailedGettingStartedLinux.pdf&quot;.</span></p>

<p class=MsoNormal style='margin-left:21.3pt;text-indent:-21.3pt'><b><span
lang=EN-US>(2)<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></b><b><span lang=EN-US>Hapl-o-MatGUI python3 source code:</span></b><span
lang=EN-US> For system requirements and installation of Hapl-o-MatGUI see the
tutorial &quot;README.md&quot; on GitHub (<a
href="https://github.com/DKMS/Hapl-o-Mat_GUI">https://github.com/DKMS/Hapl-o-Mat_GUI</a>.
In order to use Hapl-o-Mat via GUI with python3 source code, both programs have
to be installed on your device.</span></p>

<p class=MsoNormal style='margin-left:21.3pt;text-indent:-21.3pt'><b><span
lang=EN-US>(3)<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></b><b><span lang=EN-US>Hapl-o-Mat Windows binary version</span></b><span
lang=EN-US>: For installation of Hapl-o-MatGUI see the tutorial &quot;README.md&quot;
on GitHub <a href="https://github.com/DKMS/Hapl-o-Mat_WinBin">https://github.com/DKMS/Hapl-o-Mat_WinBin</a>.
The Hapl-o-Mat_WinBin version contains both, Hapl-o-Mat and the GUI.</span></p>

<p class=MsoNormal style='margin-left:21.3pt'><span lang=EN-US>&nbsp;</span></p>

<h2 style='margin-top:6.0pt;margin-right:0cm;margin-bottom:12.0pt;margin-left:
0cm'><span lang=EN-US style='font-family:"Calibri",sans-serif;font-style:normal'>Execution
of Hapl-o-Mat via GUI</span></h2>

<p class=MsoNormal><span lang=EN-US>&nbsp;</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:6.0pt;
margin-left:18.0pt;text-indent:-18.0pt'><b><span lang=EN-US style='font-size:
12.0pt;line-height:115%'>1.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></b><b><span lang=EN-US style='font-size:12.0pt;line-height:115%'>Starting
the GUI</span></b></p>

<p class=MsoNormal style='margin-bottom:6.0pt'><b><span lang=EN-US>Python
source code:</span></b><span lang=EN-US> </span><span lang=EN-US>Enter the directory
Hapl-o-MatGUI/GUIsrc in your command line interpreter and start the program via
&quot;python main.py&quot;. Please note that you require python3.x to run the
GUI.</span></p>

<p class=MsoNormal style='margin-bottom:12.0pt'><b><span lang=EN-US>Windows
binary version: </span></b><span lang=EN-US>During installation, Hapl-o-MatGUI
will be added to your start menu and can be started from there. Alternatively,
you can execute the Hapl-o-MatGUI.exe file in the </span><span lang=EN-US>directory
you chose during Hapl-o-MatGUI installation</span><span lang=EN-US>.</span></p>

<p class=MsoNormal style='margin-bottom:12.0pt'><b><span lang=EN-US>&nbsp;</span></b></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:6.0pt;
margin-left:18.0pt;text-indent:-18.0pt'><b><span lang=EN-US style='font-size:
12.0pt;line-height:115%'>2.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></b><b><span lang=EN-US style='font-size:12.0pt;line-height:115%'>Set
path to</span></b><span lang=EN-US> </span><b><span lang=EN-US
style='font-size:12.0pt;line-height:115%'>Hapl-o-Mat folder</span></b></p>

<p class=MsoNormal style='margin-bottom:6.0pt'><span lang=EN-US>To enable the
GUI to interact with Hapl-o-Mat, the path to the folder </span><span
lang=EN-US style='color:black'>containing the executable </span><span
lang=EN-US>Hapl-o-Mat</span><i><span lang=EN-US> </span></i><span lang=EN-US
style='color:black'> ('haplomat' on LinuxOS/macOS, ‘</span><span lang=EN-US>Hapl-o-Mat.exe’
on Windows OS)</span><span lang=EN-US> must be selected.</span><span
lang=EN-US style='color:black'> </span><span lang=EN-US>Button [ ... ] opens a
file dialogue. After the selection, status and actuality of the IPD-IMGT/HLA
data is automatically checked and indicated in the &quot;Update IPD-IMGT/HLA
data&quot; frame (see (2.)).</span></p>

<p class=MsoNormal style='margin-bottom:6.0pt'><b><span lang=EN-US>&nbsp;</span></b></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:6.0pt;
margin-left:18.0pt;text-indent:-18.0pt'><b><span lang=EN-US style='font-size:
12.0pt;line-height:115%'>3.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></b><b><span lang=EN-US style='font-size:12.0pt;line-height:115%'>Update
IPD-IMGT/HLA data</span></b></p>

<p class=MsoNormal style='margin-bottom:6.0pt'><span lang=EN-US>Hapl-o-Mat
haplotype frequency inference relies on information about the HLA nomenclature
provided by external files. After the first download of Hapl-o-Mat the initial
data update is mandatory. As the HLA nomenclature is constantly evolving, e.g.
by inclusion of new alleles or new multiple allele codes, it is important to
update this data regularly.</span></p>

<p class=MsoNormal style='margin-bottom:6.0pt'><span lang=EN-US>Hapl-o-Mat
relies on the following files in the folder </span><span lang=EN-US>&quot;</span><span
lang=EN-US>Hapl-o-Mat/data</span><span lang=EN-US>&quot;</span><span
lang=EN-US> that will be created by the updating process:</span></p>

<div align=center>

<table class=MsoNormalTable border=0 cellspacing=0 cellpadding=0
 style='border-collapse:collapse'>
 <tr style='height:2.85pt'>
  <td width=158 valign=top style='width:118.8pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  padding:0cm 5.4pt 0cm 5.4pt;height:2.85pt'>
  <p class=MsoNormal style='text-align:justify'><b><span lang=EN-US>File name</span></b></p>
  </td>
  <td width=406 valign=top style='width:304.8pt;border:none;border-bottom:solid windowtext 1.0pt;
  padding:0cm 5.4pt 0cm 5.4pt;height:2.85pt'>
  <p class=MsoNormal style='text-align:justify'><b><span lang=EN-US>Description</span></b></p>
  </td>
 </tr>
 <tr>
  <td width=158 valign=top style='width:118.8pt;border:none;border-right:solid windowtext 1.0pt;
  padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNoSpacing style='text-align:justify'><span lang=EN-US>AllAllelesExpanded.txt
  </span></p>
  <p class=MsoNoSpacing style='text-align:justify'><span lang=EN-US>&nbsp;</span></p>
  </td>
  <td width=406 valign=top style='width:304.8pt;border:none;padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNoSpacing style='text-align:justify'><span lang=EN-US>A list of
  relevant existing HLA alleles with their enclosed more-digit typing
  resolutions</span></p>
  </td>
 </tr>
 <tr>
  <td width=158 valign=top style='width:118.8pt;border:none;border-right:solid windowtext 1.0pt;
  padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNoSpacing style='text-align:justify'><span lang=EN-US>AlleleList.txt</span></p>
  </td>
  <td width=406 valign=top style='width:304.8pt;border:none;padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNoSpacing style='text-align:justify'><span lang=EN-US>If your
  input data in GLS format includes a missing single-locus genotype, it can be
  replaced by combining all alleles of the same locus from this file. You only must
  create it in this case.</span></p>
  </td>
 </tr>
 <tr>
  <td width=158 valign=top style='width:118.8pt;border:none;border-right:solid windowtext 1.0pt;
  padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNoSpacing style='text-align:justify'><span lang=EN-US>Ambiguity.txt</span></p>
  </td>
  <td width=406 valign=top style='width:304.8pt;border:none;padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNoSpacing style='text-align:justify'><span lang=EN-US>Data for
  the ambiguity filter</span></p>
  </td>
 </tr>
 <tr>
  <td width=158 valign=top style='width:118.8pt;border:none;border-right:solid windowtext 1.0pt;
  padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNoSpacing style='text-align:justify'><span lang=EN-US>LargeG.txt</span></p>
  </td>
  <td width=406 valign=top style='width:304.8pt;border:none;padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNoSpacing style='text-align:justify'><span lang=EN-US>A list of
  G-groups with their enclosed alleles in 8-digit resolution</span></p>
  </td>
 </tr>
 <tr style='height:12.2pt'>
  <td width=158 valign=top style='width:118.8pt;border:none;border-right:solid windowtext 1.0pt;
  padding:0cm 5.4pt 0cm 5.4pt;height:12.2pt'>
  <p class=MsoNoSpacing style='text-align:justify'><span lang=EN-US>MultipleAlleleCodes.txt</span></p>
  </td>
  <td width=406 valign=top style='width:304.8pt;border:none;padding:0cm 5.4pt 0cm 5.4pt;
  height:12.2pt'>
  <p class=MsoNoSpacing style='text-align:justify'><span lang=EN-US>A list of
  multiple allele codes and their translation to alleles in 4-digit resolution</span></p>
  </td>
 </tr>
 <tr>
  <td width=158 valign=top style='width:118.8pt;border:none;border-right:solid windowtext 1.0pt;
  padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNoSpacing style='text-align:justify'><span lang=EN-US>P.txt</span></p>
  </td>
  <td width=406 valign=top style='width:304.8pt;border:none;padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNoSpacing style='text-align:justify'><span lang=EN-US>A list of
  P-groups with their enclosed alleles in 8-digit resolution</span></p>
  </td>
 </tr>
 <tr>
  <td width=158 valign=top style='width:118.8pt;border:none;border-right:solid windowtext 1.0pt;
  padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNoSpacing style='text-align:justify'><span lang=EN-US>Smallg.txt</span></p>
  </td>
  <td width=406 valign=top style='width:304.8pt;border:none;padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNoSpacing style='text-align:justify'><span lang=EN-US>A list of
  g-groups with their enclosed alleles in 8-digit resolution</span></p>
  </td>
 </tr>
</table>

</div>

<p class=MsoNoSpacing><span lang=EN-US>&nbsp;</span></p>

<p class=MsoNormal style='margin-bottom:6.0pt'><span lang=EN-US>Starting the
GUI and selecting the Hapl-o-Mat folder automatically triggers a status and
actuality check of the data files. The date of the last update as well as the
loci available for haplotype estimation are indicated in the &quot;Update
IPD-IMGT/HLA data&quot; frame. If the data is older than three months a warning
sign and an update recommendation will be indicated. Otherwise, a green tick will
indicate that the updating step can be skipped.</span></p>

<p class=MsoNormal style='margin-bottom:6.0pt'><span lang=EN-US>The [Update]
button starts the process of downloading and automatic data processing which
may take a few minutes. A working internet connection is required for the
update process.</span></p>

<p class=MsoNormal style='margin-bottom:6.0pt'><b><span lang=EN-US>Troubleshooting
1: </span></b><span lang=EN-US>If the automatic data updating process fails,
e.g. due to </span><span lang=EN-US>a connection time out, a proxy or a
firewall issue</span><span lang=EN-US>, there is a possibility to download the
necessary data files manually, place them in the &quot;Hapl-o-Mat/prepareData&quot;
directory and then initiate the data processing via the GUI [Update] button. </span></p>

<p class=MsoNormal style='margin-bottom:6.0pt'><span lang=EN-US>The four files
and their respective locations are the following:</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:35.7pt;text-indent:-17.85pt'><span lang=EN-US>1.<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span></span><span
lang=EN-US>hla_nom_p.txt: <a href="http://hla.alleles.org/wmda/hla_nom_p.txt">http://hla.alleles.org/wmda/hla_nom_p.txt</a></span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:35.7pt;text-indent:-17.85pt'><span lang=EN-US>2.<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span></span><span
lang=EN-US>hla_nom_g.txt: <a href="http://hla.alleles.org/wmda/hla_nom_g.txt">http://hla.alleles.org/wmda/hla_nom_g.txt</a></span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:35.7pt;text-indent:-17.85pt'><span lang=EN-US>3.<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span></span><span
lang=EN-US>alpha.v3.zip: <a
href="https://bioinformatics.bethematchclinical.org/HLA/alpha.v3.zip">https://bioinformatics.bethematchclinical.org/HLA/alpha.v3.zip</a></span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:6.0pt;
margin-left:35.7pt;text-indent:-17.85pt'><span lang=EN-US>4.<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span></span><span
lang=EN-US>hla_ambigs.xml.zip: <a
href="https://github.com/jrob119/IMGTHLA/raw/Latest/xml/hla_ambigs.xml.zip">https://github.com/jrob119/IMGTHLA/raw/Latest/xml/hla_ambigs.xml.zip</a></span></p>

<p class=MsoNormal style='margin-bottom:6.0pt'><span lang=EN-US>Please make
sure to save the four files on your computer and only place a copy in
&quot;Hapl-o-Mat/prepareData&quot;, because the files will be deleted automatically
after processing.</span></p>

<p class=MsoNormal><span lang=EN-US>For detailed instructions on how to
download the necessary data files manually, please see the tutorial &quot;detailedExplanationPrepareData.doc&quot;
in the &quot;Hapl-o-Mat/ textsForGettingStarted&quot; directory. </span></p>

<p class=MsoNormal style='margin-bottom:6.0pt'><b><span lang=EN-US>Troubleshooting
2: </span></b><span lang=EN-US>After an aborted update via the [Update] button,
download files may remain undeleted in the Hapl-o-Mat subdirectory
&quot;prepareData&quot;. This will impede the download of updated data files
and is indicated by a warning message prior to the start of the download
process. In this case please abort the download and remove all following files
from the &quot;prepareData&quot; directory:</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:36.0pt;text-indent:-18.0pt'><span lang=EN-US>-<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span><span lang=EN-US>alpha.v3.zip</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:36.0pt;text-indent:-18.0pt'><span lang=EN-US>-<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span><span lang=EN-US>alpha.v3.txt</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:36.0pt;text-indent:-18.0pt'><span lang=EN-US>-<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span><span lang=EN-US>hla_ambigs.xml.zip</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:36.0pt;text-indent:-18.0pt'><span lang=EN-US>-<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span><span lang=EN-US>hla_ambigs.xml</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:36.0pt;text-indent:-18.0pt'><span lang=EN-US>-<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span><span lang=EN-US>hla_nom_g.txt</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:36.0pt;text-indent:-18.0pt'><span lang=EN-US>-<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span><span lang=EN-US>hla_nom_p.txt</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:36.0pt;text-indent:-18.0pt'><span lang=EN-US>-<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span><span lang=EN-US>Smallg.txt</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:36.0pt;text-indent:-18.0pt'><span lang=EN-US>-<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span><span lang=EN-US>LargeG.txt</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:36.0pt;text-indent:-18.0pt'><span lang=EN-US>-<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span><span lang=EN-US>P.txt</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:36.0pt;text-indent:-18.0pt'><span lang=EN-US>-<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span><span lang=EN-US>MultipleAlleleCodes.txt</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:36.0pt;text-indent:-18.0pt'><span lang=EN-US>-<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span><span lang=EN-US>allAlleles.txt</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:36.0pt;text-indent:-18.0pt'><span lang=EN-US>-<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span><span lang=EN-US>AllAllelesExpanded.txt</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:36.0pt;text-indent:-18.0pt'><span lang=EN-US>-<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span><span lang=EN-US>OneElementG.txt</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:12.0pt;
margin-left:18.0pt'><span lang=EN-US>&nbsp;</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:6.0pt;
margin-left:17.85pt;text-indent:-17.85pt'><b><span lang=EN-US style='font-size:
12.0pt;line-height:115%'>4.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></b><b><span lang=EN-US style='font-size:12.0pt;line-height:115%'>Set
run parameters</span></b></p>

<p class=MsoNormal style='margin-bottom:6.0pt'><span lang=EN-US>Run parameters
can be either set manually or loaded from an existing valid parameter file.</span></p>

<p class=MsoNormal style='margin-bottom:6.0pt'><span lang=EN-US>The
[Parameters] button opens an input mask:</span></p>

<p class=MsoNormal style='margin-bottom:6.0pt'><b><span lang=EN-US>Load
parameters: </span></b><span lang=EN-US>On the input mask, the [Load
parameters] button offers the choice of an existing parameter file via a file
dialogue. Parameters are then loaded into the respective fields of the mask and
can be edited and saved for the run.</span></p>

<p class=MsoNormal style='margin-bottom:6.0pt'><b><span lang=EN-US>Set
parameters: </span></b><span lang=EN-US>In the &quot;Set or change parameters&quot;
frame, all parameters can be set manually and saved for the run.</span></p>

<p class=MsoNormal style='margin-bottom:6.0pt'><span lang=EN-US>It is
recommended to choose a unique Run-ID for each Hapl-o-Mat run, because all
results files as well as a copy of the parameters file will be saved under
specification of the Run-ID in the chosen results folder.</span></p>

<p class=MsoNormal style='margin-bottom:6.0pt'><span lang=EN-US>&nbsp;</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:6.0pt;
margin-left:18.0pt;text-indent:-18.0pt'><b><span lang=EN-US>4.1<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp; </span></span></b><b><span
lang=EN-US>Input Genotype Data</span></b></p>

<p class=MsoNormal style='text-align:justify'><span lang=EN-US>Hapl-o-Mat
infers haplotypes from population genotype data. It supports different formats
of recording genotype data. To use Hapl-o-Mat in the GUI, your data should be
in one of the following data formats:</span></p>

<div align=center>

<table class=MsoNormalTable border=0 cellspacing=0 cellpadding=0
 style='border-collapse:collapse'>
 <tr>
  <td width=92 valign=top style='width:69.2pt;border-top:none;border-left:none;
  border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNormal><b><span lang=EN-US>Data format</span></b></p>
  </td>
  <td width=482 valign=top style='width:361.45pt;border:none;border-bottom:
  solid windowtext 1.0pt;padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNormal><b><span lang=EN-US>Description</span></b></p>
  </td>
 </tr>
 <tr>
  <td width=92 valign=top style='width:69.2pt;border:none;border-right:solid windowtext 1.0pt;
  padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNormal><span lang=EN-US>MAC</span></p>
  </td>
  <td width=482 valign=top style='width:361.45pt;border:none;padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNoSpacing style='text-align:justify'><b><span lang=EN-US>M</span></b><span
  lang=EN-US>ultiple <b>A</b>llele <b>C</b>odes: ambiguities are encoded by
  multiple allele codes (MAC). Except for the first line, input files hold an
  individual's identification number and genotype per line. Genotypes are saved
  allele by allele without locus name. Identification number and alleles are
  TAB-separated. The first line of the file is a header line indicating the
  name of the first column and the loci of the other columns. Same loci must be
  placed next to each other. For an example refer to &quot;Hapl-o-Mat/examplePopulations/populationData_a.dat&quot;.</span></p>
  </td>
 </tr>
 <tr style='height:39.7pt'>
  <td width=92 valign=top style='width:69.2pt;border:none;border-right:solid windowtext 1.0pt;
  padding:0cm 5.4pt 0cm 5.4pt;height:39.7pt'>
  <p class=MsoNormal><span lang=EN-US>GLSC</span></p>
  </td>
  <td width=482 valign=top style='width:361.45pt;border:none;padding:0cm 5.4pt 0cm 5.4pt;
  height:39.7pt'>
  <p class=MsoNoSpacing style='text-align:justify'><b><span lang=EN-US>G</span></b><span
  lang=EN-US>enotype <b>L</b>ist <b>S</b>trings <b>C</b>olumn-wise: genotypes
  with or without ambiguities are saved by genotype list strings (GLS). Input
  files hold an individual's identification number and genotype per line.
  Identification number and single-locus genotypes are TAB-separated. For an
  example refer to &quot;Hapl-o-Mat/examplePopulations/populationData_b.dat&quot;.</span></p>
  </td>
 </tr>
</table>

</div>

<p class=MsoNormal style='margin-bottom:6.0pt'><span lang=EN-US>&nbsp;</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:6.0pt;
margin-left:18.0pt;text-indent:-18.0pt'><b><span lang=EN-US>4.2<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp; </span></span></b><b><span
lang=EN-US>Parameters</span></b></p>

<p class=MsoNormal style='text-align:justify'><span lang=EN-US>Each input
format for genotype data requires a different set of parameters. The parameters
are saved in the corresponding files </span><span lang=EN-US>&quot;</span><span
lang=EN-US>parametersMAC</span><span lang=EN-US>&quot;</span><span lang=EN-US>
and </span><span lang=EN-US>&quot;</span><span lang=EN-US>parametersGLSC</span><span
lang=EN-US>&quot;</span><span lang=EN-US>.All input formats have the following
parameters in common:</span></p>

<div align=center>

<table class=MsoNormalTable border=0 cellspacing=0 cellpadding=0
 style='border-collapse:collapse'>
 <tr>
  <td width=266 valign=top style='width:199.85pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNoSpacing><b><span lang=EN-US>Parameter</span></b></p>
  </td>
  <td width=308 valign=top style='width:230.8pt;border:none;border-bottom:solid windowtext 1.0pt;
  padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNoSpacing><b><span lang=EN-US>Description</span></b></p>
  </td>
 </tr>
 <tr>
  <td width=266 valign=top style='width:199.85pt;border:none;border-right:solid windowtext 1.0pt;
  padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNoSpacing style='text-align:justify'><span lang=EN-US>FILENAME_INPUT</span></p>
  </td>
  <td width=308 valign=top style='width:230.8pt;border:none;padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNoSpacing style='text-align:justify'><span lang=EN-US>The file
  name of the input population data.</span></p>
  </td>
 </tr>
 <tr style='height:3.0pt'>
  <td width=266 valign=top style='width:199.85pt;border:none;border-right:solid windowtext 1.0pt;
  padding:0cm 5.4pt 0cm 5.4pt;height:3.0pt'>
  <p class=MsoNoSpacing><span lang=EN-US>FOLDER_RESULTS</span></p>
  </td>
  <td width=308 valign=top style='width:230.8pt;border:none;padding:0cm 5.4pt 0cm 5.4pt;
  height:3.0pt'>
  <p class=MsoNoSpacing style='text-align:justify'><span lang=EN-US>The folder
  in which the results files and a copy of the parameter file of the run are to
  be be saved.</span></p>
  </td>
 </tr>
 <tr>
  <td width=266 valign=top style='width:199.85pt;border:none;border-right:solid windowtext 1.0pt;
  padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNoSpacing><span lang=EN-US>RunID</span></p>
  </td>
  <td width=308 valign=top style='width:230.8pt;border:none;padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNormal style='margin-bottom:6.0pt'><span lang=EN-US>The
  individual run-ID for the Hapl-o-Mat run. </span></p>
  </td>
 </tr>
 <tr>
  <td width=266 valign=top style='width:199.85pt;border:none;border-right:solid windowtext 1.0pt;
  padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNoSpacing style='text-align:justify'><span lang=EN-US>LOCI_AND_RESOLUTIONS</span></p>
  </td>
  <td width=308 valign=top style='width:230.8pt;border:none;padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNoSpacing style='text-align:justify'><span lang=EN-US>Loci
  included into analysis and desired typing resolution per locus. Supported
  typing resolutions and their abbreviations are g-groups (g), P-groups (P), G-groups
  (G), 2-digit fields (1field), 4-digit fields (2field), 6-digit fields (3field),
  and 8-digit fields (4field). In order to exclude loci that are available in
  the input file, please choose the option </span><span lang=EN-US>&quot;</span><span
  lang=EN-US>ignore locus</span><span lang=EN-US>&quot;</span><span lang=EN-US>.</span></p>
  <p class=MsoNoSpacing style='text-align:justify'><span lang=EN-US>Alleles are
  not translated via the option asItIs (applying the ambiguity filter includes
  an intrinsic translation to G-groups).</span></p>
  </td>
 </tr>
 <tr style='height:43.75pt'>
  <td width=266 valign=top style='width:199.85pt;border:none;border-right:solid windowtext 1.0pt;
  padding:0cm 5.4pt 0cm 5.4pt;height:43.75pt'>
  <p class=MsoNoSpacing style='text-align:justify'><span lang=EN-US>MINIMAL_FREQUENCY_GENOTYPES</span></p>
  </td>
  <td width=308 valign=top style='width:230.8pt;border:none;padding:0cm 5.4pt 0cm 5.4pt;
  height:43.75pt'>
  <p class=MsoNoSpacing style='text-align:justify'><span lang=EN-US>Genotypes
  which split into more genotypes than the inverse of this number are discarded
  from analysis.</span></p>
  </td>
 </tr>
 <tr>
  <td width=266 valign=top style='width:199.85pt;border:none;border-right:solid windowtext 1.0pt;
  padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNoSpacing style='text-align:justify'><span lang=EN-US>DO_AMBIGUITYFILTER</span></p>
  </td>
  <td width=308 valign=top style='width:230.8pt;border:none;padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNoSpacing style='text-align:justify'><span lang=EN-US>Takes
  values </span><span lang=EN-US>&quot;</span><span lang=EN-US>true</span><span
  lang=EN-US>&quot;</span><span lang=EN-US> and </span><span lang=EN-US>&quot;</span><span
  lang=EN-US>false</span><span lang=EN-US>&quot;</span><span lang=EN-US>. The
  option </span><span lang=EN-US>&quot;</span><span lang=EN-US>true</span><span
  lang=EN-US>&quot;</span><span lang=EN-US> activates the ambiguity filter.</span></p>
  </td>
 </tr>
 <tr>
  <td width=266 valign=top style='width:199.85pt;border:none;border-right:solid windowtext 1.0pt;
  padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNoSpacing style='text-align:justify'><span lang=EN-US>EXPAND_LINES_AMBIGUITYFILTER</span></p>
  <p class=MsoNoSpacing style='text-align:justify'><span lang=EN-US>&nbsp;</span></p>
  <p class=MsoNoSpacing style='text-align:justify'><span lang=EN-US>&nbsp;</span></p>
  <p class=MsoNoSpacing style='text-align:justify'><span lang=EN-US>WRITE_GENOTYPES</span></p>
  </td>
  <td width=308 valign=top style='width:230.8pt;border:none;padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNoSpacing style='text-align:justify'><span lang=EN-US>Takes
  values </span><span lang=EN-US>&quot;</span><span lang=EN-US>true</span><span
  lang=EN-US>&quot;</span><span lang=EN-US> and </span><span lang=EN-US>&quot;</span><span
  lang=EN-US>false</span><span lang=EN-US>&quot;</span><span lang=EN-US>. If set
  to </span><span lang=EN-US>&quot;</span><span lang=EN-US>true</span><span
  lang=EN-US>&quot;</span><span lang=EN-US>, matching lines with additional
  genotype pairs in the ambiguity filter are considered.</span></p>
  <p class=MsoNoSpacing style='text-align:justify'><span lang=EN-US>Takes
  values </span><span lang=EN-US>&quot;</span><span lang=EN-US>true</span><span
  lang=EN-US>&quot;</span><span lang=EN-US> and </span><span lang=EN-US>&quot;</span><span
  lang=EN-US>false</span><span lang=EN-US>&quot;</span><span lang=EN-US>. If
  set to </span><span lang=EN-US>&quot;</span><span lang=EN-US>true</span><span
  lang=EN-US>&quot;</span><span lang=EN-US>, resolved genotypes will be written
  to a file under the path specified by FILENAME_GENOTYPES (see below).</span></p>
  </td>
 </tr>
 <tr>
  <td width=266 valign=top style='width:199.85pt;border:none;border-right:solid windowtext 1.0pt;
  padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNoSpacing><span lang=EN-US>INITIALIZATION_HAPLOTYPEFREQUENCIES</span></p>
  </td>
  <td width=308 valign=top style='width:230.8pt;border:none;padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNoSpacing style='text-align:justify'><span lang=EN-US>Initialization
  routine for haplotype frequencies. It takes the following values: </span></p>
  <p class=MsoNoSpacing style='margin-left:36.0pt;text-align:justify;
  text-indent:-18.0pt'><span lang=EN-US>-<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
  </span></span><span lang=EN-US>&quot;</span><span lang=EN-US>equal</span><span
  lang=EN-US>&quot;</span><span lang=EN-US>: All haplotype frequencies are
  initialized with the same frequency, the inverse number of haplotypes.</span></p>
  <p class=MsoNoSpacing style='margin-left:36.0pt;text-align:justify;
  text-indent:-18.0pt'><span lang=EN-US>-<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
  </span></span><span lang=EN-US>&quot;numberOccurrence&quot;: Haplotype
  frequencies are initialized according to the initial number of occurrence of
  haplotypes.</span></p>
  <p class=MsoNoSpacing style='margin-left:36.0pt;text-align:justify;
  text-indent:-18.0pt'><span lang=EN-US>-<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
  </span></span><span lang=EN-US>&quot;random&quot;: Haplotype frequencies are
  initialized randomly.</span></p>
  <p class=MsoNoSpacing style='margin-left:36.0pt;text-align:justify;
  text-indent:-18.0pt'><span lang=EN-US>-<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
  </span></span><span lang=EN-US>&quot;perturbation&quot;: Haplotype
  frequencies are initialized as in numberOccurrence and then randomly modified
  by a small (&lt;10%) positive or negative offset.</span></p>
  </td>
 </tr>
 <tr>
  <td width=266 valign=top style='width:199.85pt;border:none;border-right:solid windowtext 1.0pt;
  padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNoSpacing><span lang=EN-US>EPSILON</span></p>
  </td>
  <td width=308 valign=top style='width:230.8pt;border:none;padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNoSpacing style='text-align:justify'><span lang=EN-US>Value for
  the stopping criterion, i.e. the maximal change between consecutive haplotype
  frequency estimations is smaller than the assigned value. Values larger than
  1/(2*<i>n</i>), with <i>n</i> being the number of genotypes in the input
  file, are not valid.</span></p>
  </td>
 </tr>
 <tr>
  <td width=266 valign=top style='width:199.85pt;border:none;border-right:solid windowtext 1.0pt;
  padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNoSpacing><span lang=EN-US>CUT_HAPLOTYPEFREQUENCIES</span></p>
  </td>
  <td width=308 valign=top style='width:230.8pt;border:none;padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNoSpacing style='text-align:justify'><span lang=EN-US>Estimated
  haplotype frequencies smaller than this value are removed from the output.</span></p>
  </td>
 </tr>
 <tr>
  <td width=266 valign=top style='width:199.85pt;border:none;border-right:solid windowtext 1.0pt;
  padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNoSpacing><span lang=EN-US>RENORMALIZE_HAPLOTYPEFREQUENCIES</span></p>
  </td>
  <td width=308 valign=top style='width:230.8pt;border:none;padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNoSpacing style='text-align:justify'><span lang=EN-US>Takes
  values &quot;true&quot; and &quot;false&quot;. If &quot;true&quot;, normalize
  estimated haplotype frequencies to sum to one. Within machine precision, this
  becomes necessary, if estimated haplotypes are removed, e.g. via the option
  CUT_HAPLOTYPEFREQUENCIES.</span></p>
  </td>
 </tr>
 <tr>
  <td width=266 valign=top style='width:199.85pt;border:none;border-right:solid windowtext 1.0pt;
  padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNoSpacing><span lang=EN-US>SEED</span></p>
  </td>
  <td width=308 valign=top style='width:230.8pt;border:none;padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNoSpacing style='text-align:justify'><span lang=EN-US>Set the
  seed of the used pseudo random number generator. If set to &quot;0&quot;, the
  seed is initialized by the system time. Valid entries are integer values
  between 0 and 1,000,000,000.</span></p>
  </td>
 </tr>
</table>

</div>

<p class=MsoNoSpacing><span lang=EN-US>&nbsp;</span></p>

<p class=MsoNoSpacing><span lang=EN-US>The following parameters are generated
automatically upon entry of results folder and RunID:</span></p>

<p class=MsoNoSpacing><span lang=EN-US>&nbsp;</span></p>

<div align=center>

<table class=MsoNormalTable border=0 cellspacing=0 cellpadding=0
 style='border-collapse:collapse'>
 <tr>
  <td width=266 valign=top style='width:199.85pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNoSpacing><b><span lang=EN-US>Parameter</span></b></p>
  </td>
  <td width=308 valign=top style='width:230.8pt;border:none;padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNoSpacing style='text-align:justify'><b><span lang=EN-US>Description</span></b></p>
  </td>
 </tr>
 <tr>
  <td width=266 valign=top style='width:199.85pt;border:none;border-right:solid windowtext 1.0pt;
  padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNoSpacing><span lang=EN-US>FILENAME_HAPLOTYPES</span></p>
  </td>
  <td width=308 valign=top style='width:230.8pt;border:none;border-top:solid windowtext 1.0pt;
  padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNoSpacing style='text-align:justify'><span lang=EN-US>Name of the
  file which temporarily saves haplotype names</span></p>
  </td>
 </tr>
 <tr>
  <td width=266 valign=top style='width:199.85pt;border:none;border-right:solid windowtext 1.0pt;
  padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNoSpacing><span lang=EN-US>FILENAME_GENOTYPES</span></p>
  </td>
  <td width=308 valign=top style='width:230.8pt;border:none;padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNoSpacing style='text-align:justify'><span lang=EN-US>Name of the
  file which saves resolved genotypes</span></p>
  </td>
 </tr>
 <tr>
  <td width=266 valign=top style='width:199.85pt;border:none;border-right:solid windowtext 1.0pt;
  padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNoSpacing><span lang=EN-US>FILENAME_HAPLOTYPEFREQUENCIES</span></p>
  </td>
  <td width=308 valign=top style='width:230.8pt;border:none;padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNoSpacing style='text-align:justify'><span lang=EN-US>Name of the
  file which saves haplotypes and estimated haplotype frequencies</span></p>
  </td>
 </tr>
 <tr>
  <td width=266 valign=top style='width:199.85pt;border:none;border-right:solid windowtext 1.0pt;
  padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNoSpacing><span lang=EN-US>FILENAME_EPSILON_LOGL</span></p>
  </td>
  <td width=308 valign=top style='width:230.8pt;border:none;padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNoSpacing style='text-align:justify'><span lang=EN-US>Name of the
  file which saves stopping criterion and log-likelihood per iteration</span></p>
  </td>
 </tr>
</table>

</div>

<p class=MsoNoSpacing><span lang=EN-US>&nbsp;</span></p>

<p class=MsoNoSpacing><span lang=EN-US>&nbsp;</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:6.0pt;
margin-left:18.0pt;text-indent:-18.0pt'><b><span lang=EN-US style='font-size:
12.0pt;line-height:115%'>5<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></b><b><span lang=EN-US style='font-size:12.0pt;line-height:115%'>Run
Hapl-o-Mat</span></b></p>

<p class=MsoNormal style='margin-bottom:6.0pt'><span lang=EN-US>The [Run] button
starts the Hapl-o-Mat run. Hapl-o-Mat terminal output is displayed.
Simultaneously with the EM algorithm the current course of the epsilon value is
displayed in the right plot of the &quot;results&quot; window. When the run is
finished, the terminal output prints &quot;Finished!&quot; and displays the
time stamp. Results are displayed in the &quot;Results&quot; frame.</span></p>

<p class=MsoNormal style='margin-bottom:6.0pt'><span lang=EN-US>All terminal
output of Hapl-o-Mat is saved as &quot;RunID_Log.dat&quot; in the chosen results
folder.</span></p>

<p class=MsoNormal style='margin-bottom:6.0pt'><span lang=EN-US>&nbsp;</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:6.0pt;
margin-left:18.0pt;text-indent:-18.0pt'><b><span lang=EN-US style='font-size:
12.0pt;line-height:115%'>6<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></b><b><span lang=EN-US style='font-size:12.0pt;line-height:115%'>Results</span></b></p>

<p class=MsoNormal style='margin-bottom:6.0pt'><span lang=EN-US>When the
Hapl-o-Mat run is finished, result files are saved and results are
automatically displayed in the &quot;Results&quot; frame.</span></p>

<p class=MsoNormal style='margin-bottom:6.0pt'><span lang=EN-US>&nbsp;</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:6.0pt;
margin-left:18.0pt;text-indent:-18.0pt'><b><span lang=EN-US>6.1<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp; </span></span></b><b><span
lang=EN-US>Result files</span></b></p>

<p class=MsoNormal style='margin-bottom:6.0pt'><span lang=EN-US>All results
files will be found in the specified results folder with a tag of the chosen
RunID. </span></p>

<p class=MsoNormal style='margin-bottom:6.0pt'><span lang=EN-US>Results files
are:</span></p>

<div align=center>

<table class=MsoNormalTable border=0 cellspacing=0 cellpadding=0
 style='border-collapse:collapse'>
 <tr>
  <td width=99 valign=top style='width:74.35pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNoSpacing><b><span lang=EN-US>File type</span></b></p>
  </td>
  <td width=161 valign=top style='width:120.5pt;border:none;border-right:solid windowtext 1.0pt;
  padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNoSpacing style='text-align:justify'><b><span lang=EN-US>File
  name</span></b></p>
  </td>
  <td width=354 valign=top style='width:265.6pt;border:none;padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNoSpacing style='text-align:justify'><b><span lang=EN-US>Description</span></b></p>
  </td>
 </tr>
 <tr>
  <td width=99 valign=top style='width:74.35pt;border:none;border-right:solid windowtext 1.0pt;
  padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNoSpacing><span lang=EN-US>Parameters</span></p>
  </td>
  <td width=161 valign=top style='width:120.5pt;border-top:solid windowtext 1.0pt;
  border-left:none;border-bottom:none;border-right:solid windowtext 1.0pt;
  padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNoSpacing style='text-align:justify'><i><span lang=EN-US>RunID</span></i><span
  lang=EN-US>_parametersGLSC or <i>RunID</i>_parametersMAC</span></p>
  </td>
  <td width=354 valign=top style='width:265.6pt;border:none;border-top:solid windowtext 1.0pt;
  padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNoSpacing style='text-align:justify'><span lang=EN-US>A copy of
  the Hapl-o-Mat parameter file created in the &quot;Set parameters&quot; step.
  The extension changes between &quot;GLSC&quot; or &quot;MAC&quot; depending
  on the format of the input file. These parameter files can be loaded and
  edited under &quot;Set parameters&quot; for subsequent runs.</span></p>
  </td>
 </tr>
 <tr>
  <td width=99 valign=top style='width:74.35pt;border:none;border-right:solid windowtext 1.0pt;
  padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNoSpacing><span lang=EN-US>Haplotype frequencies</span></p>
  </td>
  <td width=161 valign=top style='width:120.5pt;border:none;border-right:solid windowtext 1.0pt;
  padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNoSpacing style='text-align:justify'><i><span lang=EN-US>RunID</span></i><span
  lang=EN-US>_htf.dat</span></p>
  </td>
  <td width=354 valign=top style='width:265.6pt;border:none;padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNoSpacing style='text-align:justify'><span lang=EN-US>Complete
  list of haplotypes and their frequencies as</span></p>
  <p class=MsoNoSpacing style='text-align:justify'><span lang=EN-US>estimated
  from the input genotypes.</span></p>
  </td>
 </tr>
 <tr>
  <td width=99 valign=top style='width:74.35pt;border:none;border-right:solid windowtext 1.0pt;
  padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNoSpacing><span lang=EN-US>Log</span></p>
  </td>
  <td width=161 valign=top style='width:120.5pt;border:none;border-right:solid windowtext 1.0pt;
  padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNoSpacing style='text-align:justify'><i><span lang=EN-US>RunID</span></i><span
  lang=EN-US>_Log.dat</span></p>
  </td>
  <td width=354 valign=top style='width:265.6pt;border:none;padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNoSpacing style='text-align:justify'><span lang=EN-US>Copy of
  Hapl-o-Mat terminal output.</span></p>
  </td>
 </tr>
 <tr>
  <td width=99 valign=top style='width:74.35pt;border:none;border-right:solid windowtext 1.0pt;
  padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNoSpacing><span lang=EN-US>Epsilon log</span></p>
  </td>
  <td width=161 valign=top style='width:120.5pt;border:none;border-right:solid windowtext 1.0pt;
  padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNoSpacing style='text-align:justify'><i><span lang=EN-US>RunID</span></i><span
  lang=EN-US>_epsilon.dat</span></p>
  </td>
  <td width=354 valign=top style='width:265.6pt;border:none;padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNoSpacing style='text-align:justify'><span lang=EN-US>Log of the
  changes in stopping criterion Epsilon and the log-likelihood during the
  iterating expectation and maximization steps. The first column is the
  stopping criterion Epsilon and the second one the not normalized
  log-likelihood.</span></p>
  </td>
 </tr>
 <tr>
  <td width=99 valign=top style='width:74.35pt;border:none;border-right:solid windowtext 1.0pt;
  padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNoSpacing><span lang=EN-US>Genotypes</span></p>
  </td>
  <td width=161 valign=top style='width:120.5pt;border:none;border-right:solid windowtext 1.0pt;
  padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNoSpacing style='text-align:justify'><i><span lang=EN-US>RunID</span></i><span
  lang=EN-US>_genotypes.dat</span></p>
  </td>
  <td width=354 valign=top style='width:265.6pt;border:none;padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=MsoNormal style='text-align:justify'><span lang=EN-US>Information on
  how the input genotypes were resolved for haplotype estimations. The first
  column corresponds to the individual's identification number. The second
  column indicates how ambiguities per single-locus genotypes have been
  resolved. If no ambiguity occurred or no additional genotypes are formed, the
  type is N. If an ambiguity occurred and was resolved via building all
  possible allele combinations, the type is I. Activating the ambiguity filter
  gives additional types: A, if one matching line in the ambiguity file was
  found, and M if multiple matching lines were found. The third column gives
  the frequency of the genotype and the fourth column the genotype itself. The
  genotype is saved in the GLS format. If an individual's genotype splits into
  a set of genotypes, each genotype is written to one line starting with the
  same identification number. The corresponding frequencies become non-integer
  and sum to one.</span></p>
  </td>
 </tr>
</table>

</div>

<p class=MsoNormal style='margin-bottom:6.0pt'><span lang=EN-US>&nbsp;</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:6.0pt;
margin-left:18.0pt;text-indent:-18.0pt'><b><span lang=EN-US>6.2<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp; </span></span></b><b><span
lang=EN-US>Result display</span></b></p>

<p class=MsoNormal style='margin-bottom:6.0pt'><span lang=EN-US>The results
display mainly consists of a list of the haplotypes and their estimated frequencies,
the plotted haplotype frequencies and the plotted epsilon values of the EM
algorithm.</span></p>

<p class=MsoNormal style='margin-bottom:6.0pt'><span lang=EN-US>The top line
&quot;Statistics&quot; of the &quot;Results&quot; frame provides information on</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:6.0pt;
margin-left:36.0pt;text-indent:-18.0pt'><span lang=EN-US>-<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span><span lang=EN-US>the <b>total number of haplotypes</b> in the
results file, </span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:6.0pt;
margin-left:36.0pt;text-indent:-18.0pt'><span lang=EN-US>-<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span><span lang=EN-US>the <b>number of genotypes (<i>n</i>)</b> that
contributed to the results (e.g. genotypes which were discarded from analysis
because of excess ambiguity are not counted), </span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:6.0pt;
margin-left:36.0pt;text-indent:-18.0pt'><span lang=EN-US>-<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span><span lang=EN-US>the critical value <b>1/(2*<i>n</i>)</b> which
specifies the theoretically smallest possible frequency of a haplotype represented
once in the sample,</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:6.0pt;
margin-left:36.0pt;text-indent:-18.0pt'><span lang=EN-US>-<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span><span lang=EN-US>and the <b>sum of the cut haplotype frequencies</b>
(which will be 1 if none of the haplotypes were cut or if the frequencies were
normalized).</span></p>

<p class=MsoNormal style='margin-bottom:6.0pt'><span lang=EN-US>&nbsp;</span></p>

<p class=MsoNormal style='margin-bottom:6.0pt'><span lang=EN-US>The line below
(&quot;Display&quot;) offers different options to limit the number of
haplotypes displayed in the list view and the frequency plot. The options are</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:6.0pt;
margin-left:36.0pt;text-indent:-18.0pt'><span lang=EN-US>-<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span><b><span lang=EN-US>Top</span></b><span lang=EN-US>: display of
the Top <i>x</i> haplotypes (<i>x </i>can be set manually)</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:6.0pt;
margin-left:36.0pt;text-indent:-18.0pt'><span lang=EN-US>-<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span><b><span lang=EN-US>All</span></b><span lang=EN-US>: all
haplotypes in the Hapl-o-Mat output as defined by parameter CUT_HAPLOTYPEFREQUENCIES.</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:6.0pt;
margin-left:36.0pt;text-indent:-18.0pt'><span lang=EN-US>-<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span><b><span lang=EN-US>All haplotypes with frequency &#8805; 1/(2*<i>n</i>)</span></b><span
lang=EN-US>: with n = the number of genotypes<b> </b>that contributed to the
results, as indicated in the &quot;Statistics&quot; line.</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:6.0pt;
margin-left:36.0pt;text-indent:-18.0pt'><span lang=EN-US>-<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span><b><span lang=EN-US>All haplotypes with cumulated frequency
&#8804; <i>y</i></span></b><span lang=EN-US>: <i>y </i>can be set manually</span></p>

<p class=MsoNormal style='margin-bottom:6.0pt'><span lang=EN-US>Values from the
haplotype frequencies list can be copied (STRG+A, STRG+C) to clipboard.</span></p>

<p class=MsoNormal style='margin-bottom:6.0pt'><span lang=EN-US>The axes of the
two plots can be scaled logarithmically by clicking the options on the bars
below the plots.</span></p>

<p class=MsoNormal style='margin-bottom:6.0pt'><span lang=EN-US>&nbsp;</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:6.0pt;
margin-left:18.0pt;text-indent:-18.0pt'><b><span lang=EN-US style='font-size:
12.0pt;line-height:115%'>7<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></b><b><span lang=EN-US style='font-size:12.0pt;line-height:115%'>Unsolved
Issues </span></b></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:6.0pt;
margin-left:18.0pt;text-indent:-18.0pt'><b><span lang=EN-US>7.1<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp; </span></span></b><b><span
lang=EN-US>Plot scaling issue:</span></b><span lang=EN-US> Plot scaling
adjustment is incorrect when switching between monitors with different DPI
settings. This issue is a known PyQtGraph bug that has not been resolved yet (see
e.g.: https://github.com/pyqtgraph/pyqtgraph/issues/756).</span></p>

<p class=MsoNormal style='margin-bottom:6.0pt'><span lang=EN-US>&nbsp;</span></p>

</div>

</body>

</html>
